### PR TITLE
Kiali-1553 upgrade React library to 16.5.2

### PR DESCRIPTION
**Pull request by vnugent**
** Describe the change **

- Upgraded major react-* and corresponding types libs
- Converted React.PropTypes to Typescript

** Issue reference **
KIALI-1553

** Backwards compatible? **
Yes

### CODE REVIEW

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-ui-pr-e2e-test [#221](https://paste.fedoraproject.org/paste/wfhWSLqTOIRHD3JTgePTtQ)
  * :red_circle: Istio-and-Kiali-on-Openshift [#[1632]](https://paste.fedoraproject.org/paste/mC5jsIbXbw7C2fNj4lNFKg)
  * :heavy_check_mark: Istio-Upstream [#[480]](https://paste.fedoraproject.org/paste/~ng0Z0Rh1xN8MQDMXljzmg)
  * :heavy_check_mark: Kiali-Istio-Checker [#[1830, 1831]](https://paste.fedoraproject.org/paste/bAZY66n8IGfSe55zp13RQA)
  * :heavy_check_mark: install-bookinfo [#[1007]](https://paste.fedoraproject.org/paste/R76GbPuBfZvgpF3l-jhpkg)
  * :heavy_check_mark: Kiali-on-Openshift [#[1023]](https://paste.fedoraproject.org/paste/GOBfzBurq9UR6-JNDEuG0g)
  * :red_circle: run-kiali-e2e-tests [#[429]](https://paste.fedoraproject.org/paste/II9Lp9wxlNRoZ~HnGHCjGg)


KIALI_NAME=kialiuipr753-istio-system

File: yarn.lock [hunk 1]
```diff
@@ -5,14 +5,12 @@
 "@babel/code-frame@^7.0.0-beta.35":
   version "7.0.0-beta.47"
   resolved "https://registry.yarnpkg.com/@babel/code-frame/-/code-frame-7.0.0-beta.47.tgz#d18c2f4c4ba8d093a2bcfab5616593bfe2441a27"
-  integrity sha512-W7IeG4MoVf4oUvWfHUx9VG9if3E0xSUDf1urrnNYtC2ow1dz2ptvQ6YsJfyVXDuPTFXz66jkHhzMW7a5Eld7TA==
```
Comment (josejulio) [inline yarn.lock]: Why is the integrity removed? It happens on -I think- all the dependencies.

Are you using an older version of yarn?

**Comment by mtho11** [comment]
Just added the DNM since we want to add this after the release

**Comment by vnugent** [comment]
Upon upgrading react-dom and react-router I'm currently blocked by this compilation error.  I find it a bit difficult to understand the relationship `Overviewpage` and `ListPage.Component`. 
```
src/pages/Overview/OverviewPage.tsx
(75,42): Type 'OverviewProps' does not satisfy the constraint '{ setActiveNamespace?: string | undefined; }'.
  Types of property 'setActiveNamespace' are incompatible.
    Type '(name: Namespace) => (namespace: Namespace) => void' is not assignable to type 'string'.
```
I'm going to try to undo `react-router` upgrade.

File: yarn.lock (hunk 1 above)
Comment (vnugent) [reply yarn.lock]: I think it was caused by an older version of Yarn.  I'll address it in the next update.

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-ui-pr-e2e-test [#235](https://paste.fedoraproject.org/paste/AbVDJBnmUf7Mv~dQbEsxKA)
  * :red_circle: Istio-and-Kiali-on-Openshift [#[1685]](https://paste.fedoraproject.org/paste/JFV4FS0NJrnu7GwIf6-EDA)
  * :heavy_check_mark: Istio-Upstream [#[530]](https://paste.fedoraproject.org/paste/rnI-Cdqd2HkQuLpkdUElKg)
  * :heavy_check_mark: Kiali-Istio-Checker [#[1946]](https://paste.fedoraproject.org/paste/WtDmwZsmCKgCAnQ8n~8PGw)
  * :heavy_check_mark: install-bookinfo [#[1060]](https://paste.fedoraproject.org/paste/QWg6PnMpi1Ok4h8VsXTyAw)
  * :red_circle: Kiali-on-Openshift [#[1076]](https://paste.fedoraproject.org/paste/tnPotwxSaoutcIx1H6EcjQ)


KIALI_NAME=kialiuipr753-istio-system

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-ui-pr-e2e-test [#248](https://paste.fedoraproject.org/paste/cgVVvi~ob2VzLtWVr1CpYw)
  * :red_circle: Istio-and-Kiali-on-Openshift [#[1703]](https://paste.fedoraproject.org/paste/IKcEQZRHHDS0PAHO~1CyDw)
  * :heavy_check_mark: Istio-Upstream [#[548]](https://paste.fedoraproject.org/paste/~75O8hq47AYPF51DCvBHrw)
  * :heavy_check_mark: Kiali-Istio-Checker [#[1984, 1983]](https://paste.fedoraproject.org/paste/46xw0nIqQgFhTjpt-tXT7g)
  * :heavy_check_mark: install-bookinfo [#[1079]](https://paste.fedoraproject.org/paste/QN-Q7aVLmmJi98lzyRCnkA)
  * :heavy_check_mark: Kiali-on-Openshift [#[1094]](https://paste.fedoraproject.org/paste/w98kQX1XdWPbS85BQ7YkPg)
  * :red_circle: run-kiali-e2e-tests [#[503]](https://paste.fedoraproject.org/paste/H5ndzRrMl67GpUNPNeqYfA)


KIALI_NAME=kialiuipr753-istio-system

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-ui-pr-e2e-test [#249](https://paste.fedoraproject.org/paste/TtiPYdYdriGW6G7hReXKUw)
  * :heavy_check_mark: Istio-and-Kiali-on-Openshift [#[1705]](https://paste.fedoraproject.org/paste/Ed61STLUGErrFKY2misTCw)
  * :heavy_check_mark: Istio-Upstream [#[550]](https://paste.fedoraproject.org/paste/Q9XSo9Emo-2pOde8jO-bwA)
  * :heavy_check_mark: Kiali-Istio-Checker [#[1988, 1989]](https://paste.fedoraproject.org/paste/Omn8Os4JMy2VI2BOwEGZrg)
  * :heavy_check_mark: install-bookinfo [#[1081]](https://paste.fedoraproject.org/paste/mzREBZ-RqzkV7KNvSEzjbg)
  * :heavy_check_mark: Kiali-on-Openshift [#[1097]](https://paste.fedoraproject.org/paste/t0R0H1OIHQFbxLbpwvBYOQ)
  * :heavy_check_mark: run-kiali-e2e-tests [#[505]](https://paste.fedoraproject.org/paste/RY~BLvCpA41f2SkUmPz4hw)


KIALI_NAME=kialiuipr753-istio-system

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-ui-pr-e2e-test [#250](https://paste.fedoraproject.org/paste/K~38wezTb5rlMb9PuKByVw)
  * :heavy_check_mark: Istio-and-Kiali-on-Openshift [#[1707]](https://paste.fedoraproject.org/paste/iBBNap7L8JRZ1NxddtHtAg)
  * :heavy_check_mark: Istio-Upstream [#[552]](https://paste.fedoraproject.org/paste/t0rZUmBOq9ytjnXC~9CEEw)
  * :heavy_check_mark: Kiali-Istio-Checker [#[1992, 1993]](https://paste.fedoraproject.org/paste/T93Kmvum1aUziW3QD7IK4Q)
  * :heavy_check_mark: install-bookinfo [#[1083]](https://paste.fedoraproject.org/paste/5AYLZpsXYEAgxxP8t-fs8w)
  * :heavy_check_mark: Kiali-on-Openshift [#[1099]](https://paste.fedoraproject.org/paste/ovwSydAEePm0ZYwyMuRZ2A)
  * :heavy_check_mark: run-kiali-e2e-tests [#[507]](https://paste.fedoraproject.org/paste/TXjHSg9jzkIlRv9kqYR72A)


KIALI_NAME=kialiuipr753-istio-system

**Pull request merged**